Streetcode/Streetcode.BLL/MediatR/Authentication/Login/LoginHandler.cs

```diff
@@ -0,0 +1,62 @@
+﻿using System.IdentityModel.Tokens.Jwt;
+using AutoMapper;
+using FluentResults;
+using MediatR;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.Extensions.Localization;
+using Streetcode.BLL.DTO.Authentication.Login;
+using Streetcode.BLL.DTO.Users;
+using Streetcode.BLL.Interfaces.Logging;
+using Streetcode.BLL.Interfaces.Authentication;
+using Streetcode.DAL.Entities.Users;
+using Streetcode.DAL.Repositories.Interfaces.Base;
+
+namespace Streetcode.BLL.MediatR.Authentication.Login
+{
+    public class LoginHandler : IRequestHandler<LoginQuery, Result<LoginResponseDTO>>
+    {
+        private readonly IMapper _mapper;
+        private readonly IRepositoryWrapper _repositoryWrapper;
+        private readonly ITokenService _tokenService;
+        private readonly ILoggerService _logger;
+        private readonly UserManager<User> _userManager;
+
+        public LoginHandler(IRepositoryWrapper repositoryWrapper, IMapper mapper, ITokenService tokenService, ILoggerService logger, UserManager<User> userManager)
+        {
+            _repositoryWrapper = repositoryWrapper;
+            _mapper = mapper;
+            _tokenService = tokenService;
+            _logger = logger;
+            _userManager = userManager;
+        }
+
+        public async Task<Result<LoginResponseDTO>> Handle(LoginQuery request, CancellationToken cancellationToken)
+        {
+            var user = await _repositoryWrapper.UserRepository
+               .GetFirstOrDefaultAsync(user => user.UserName == request.UserLogin.Login || user.Email == request.UserLogin.Login);

```

Comment:
The line is too long (we can afford long lines for class constructor, but the business logic of function should be well readable )